client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/query/RemoteQuery.java

```diff
@@ -34,13 +36,23 @@
       super(queryFactory, queryString);
       this.cache = cache;
       this.serializationContext = serializationContext;
+      this.indexedQueryMode = IndexedQueryMode.FETCH;
+   }
+
+   RemoteQuery(QueryFactory queryFactory, RemoteCacheImpl<?, ?> cache, SerializationContext serializationContext,
```

Comment:
This newly added constructor should be used in RemoteQueryFactory.create(String queryString), and the old constructor (without IndexedQueryMode) can be removed.